logos/benchmarks/benchmark_logos.py

```
@@ -0,0 +1,1495 @@
+                    return None
+                total += latest.energy_mj
+        return total
+
+    def energy_from_counter(self, start_mj: float, end_mj: float) -> float:
+        return (end_mj - start_mj) / 1000.0
+
+    def energy_from_samples(self, t_start: float, t_end: float) -> Optional[float]:
+        """Trapezoidal integration of total power across all hosts → Joules."""
+        # Build a merged (local_monotonic_t, total_mw) series
+        combined: list[tuple[float, float]] = []
+        for idx, (samples, lock) in enumerate(zip(self._all_samples, self._locks)):
```

Ruff (0.15.15)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)